### PR TITLE
Update pytype_runner.py - change escape_ninja_path to escape all the ninja path escape-needing characters

### PR DESCRIPTION
**Pull request by wyattscarpenter**
This fixes https://github.com/google/pytype/issues/499

Unfortunately, I was not able to get pytype running on my machine, so this commit is just based on what seems like the right idea to me, and hopefully it can be verified to be correct by a maintainer and then merged.

### CODE REVIEW

File: pytype/tools/analyze_project/pytype_runner.py [hunk 1]
```diff
@@ -146,19 +146,31 @@ def _module_to_output_path(mod):
...
+  Escape colon, dollar sign, space, and new line, for ninja
+  (as described in https://ninja-build.org/manual.html#ref_lexer).
+  This function should only ever be used once, called on a path string to turn
+  it into a ninja path string. (If you call it on a ninja path string, it will
+  render the variables inert.)
+
+  Args:
+    path: The path.
+  """
+  new_path = ''
```
Comment (rchen152) [inline pytype/tools/analyze_project/pytype_runner.py:160]: Now that there are multiple special characters involved, I'd prefer to use `re.sub` to make the code more concise. This seems to work:
```
return re.sub(r'(?P<char>[\n :$])', r'$\g<char>', path)
```
(An `import re` line is also needed at the top of the file, of course.)

File: pytype/tools/analyze_project/pytype_runner.py [hunk 2]
```diff
@@ -338,7 +350,7 @@ def write_build_statement(self, module, action, deps, imports, suffix):
     logging.info('%s %s\n  imports: %s\n  deps: %s\n  output: %s',
                  action, module.name, imports, deps, output)
     if deps:
-      deps = ' | ' +  escape_ninja_path(' '.join(deps))
+      deps = ' | ' + ' '.join([escape_ninja_path(dep) for dep in deps])
```
Comment (rchen152) [inline pytype/tools/analyze_project/pytype_runner.py:353]:
```suggestion
      deps = ' | ' + ' '.join(escape_ninja_path(dep) for dep in deps)
```

**Review by rchen152 — CHANGES_REQUESTED** [review]
Thanks! A couple suggestions to make the code more concise.

File: pytype/tools/analyze_project/pytype_runner.py (hunk 1 above)
Comment (wyattscarpenter) [reply pytype/tools/analyze_project/pytype_runner.py]: `re.sub(r'([\n :$])', r'$\1', path)` also works equivalently, and reads slightly simpler, if you prefer that.

**Review by rchen152 — APPROVED** [review]

**Pull request merged**